lib/bolt/cli.rb

```diff
@@ -559,6 +563,38 @@ def initialize_project
       ok ? 0 : 1
     end
 
+    def migrate_project
+      if inventory.version == 2
+        ok = true
+      else
+        inventoryfile = config.inventoryfile || config.boltdir.inventory_file
+        inv = YAML.safe_load(File.open(inventoryfile))
+
+        inv = Bolt::Util.walk_pairs(inv) do |k, v|
```

Comment:
It would be helpful to have comments explicitly documenting the set of transformations we make here.

Reply:
It might be clearer to do this with explicit recursion than with a helper function.

```rb
def migrate_group(group)
  if group.key?('nodes')
    targets = group['nodes'].map do |target|
      target['uri'] = target.delete('name') if target.is_a?(Hash)
      target
    end
    group.delete('nodes')
    group['targets'] = targets
  end
  (group['groups'] || {}).map { |name, subgroup| migrate_group(subgroup) }
end
```